crates/turborepo-lib/src/ui/mod.rs

```diff
@@ -35,12 +37,44 @@ impl UI {
         // with the style when the object is displayed.
         obj.force_styling(!self.should_strip_ansi)
     }
+
+    // Ported from Go code. Converts an index to a color along the rainbow
+    fn rainbow_rgb(i: usize) -> (u8, u8, u8) {
+        let f = 0.275;
+        let r = (f * i as f64 + 4.0 * PI / 3.0).sin() * 127.0 + 128.0;
+        let g = 45.0;
+        let b = (f * i as f64).sin() * 127.0 + 128.0;
+
+        (r as u8, g as u8, b as u8)
+    }
+
+    pub fn rainbow<'a>(&self, text: &'a str) -> Cow<'a, str> {
+        if self.should_strip_ansi {
+            return Cow::Borrowed(text);
+        }
+
+        let mut out = Vec::new();
+        for (i, c) in text.char_indices() {
+            let (r, g, b) = Self::rainbow_rgb(i);
+            out.push(format!(
+                "\x1b[1m\x1b[38;2;{};{};{}m{}\x1b[0m\x1b[0;1m",
+                r, g, b, c
+            ));
+        }
+        out.push(RESET.to_string());
```

Comment:
Not blocking, but we use [`tiny_gradient`](https://github.com/vercel/turbo/blob/main/crates/turborepo-lib/src/shim.rs#L484) in the shim for coloring. It would be nice to consolidate our gradient logic

Reply:
Yeah...I'm personally of the opinion that we should keep this because it has a fallback for Apple Terminal and has color stripping.